linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3153,41 +3137,8 @@ private[optimizer] abstract class OptimizerCore(
 
       // java.lang.Class
 
-      case ClassGetComponentType =>
-        optTReceiver.get match {
-          case PreTransLit(ClassOf(ArrayTypeRef(base, depth))) =>
-            contTree(ClassOf(
-                if (depth == 1) base
-                else ArrayTypeRef(base, depth - 1)))
-          case PreTransLit(ClassOf(ClassRef(_))) =>
-            contTree(Null())
-          case receiver =>
-            default
-        }
-
       case ClassGetName =>
         optTReceiver.get match {
-          case PreTransMaybeBlock(bindingsAndStats, PreTransLit(ClassOf(typeRef))) =>
-            def mappedClassName(className: ClassName): String = {
-              RuntimeClassNameMapperImpl.map(
-                  config.coreSpec.semantics.runtimeClassNameMapper,
-                  className.nameString)
-            }
-
-            val nameString = typeRef match {
-              case primRef: PrimRef =>
-                primRef.displayName
-              case ClassRef(className) =>
-                mappedClassName(className)
-              case ArrayTypeRef(primRef: PrimRef, dimensions) =>
-                "[" * dimensions + primRef.charCode
-              case ArrayTypeRef(ClassRef(className), dimensions) =>
-                "[" * dimensions + "L" + mappedClassName(className) + ";"
-            }
-
-            contTree(finishTransformBindings(
-                bindingsAndStats, StringLiteral(nameString)))
-
```

Comment:
Yes, but also `jl.Class` would always be considered to be instantiated, which would cancel some of the benefits of having `ObjectClassName` at all. Currently this optimization allows `jl.Class` to be entirely dead-code-eliminated if neither `GetClass` nor `ClassOf` are used.

(but perhaps if it was a hijacked class, it would not produce so much more code anyway; not sure)

Reply:
No actually that's not even true, because `GetClass` returns a nullable `jl.Class`. So it would still be reified at least for a `CheckNotNull` before continuing to `Class_name`.